extensions/spring-data-rest/deployment/src/main/java/io/quarkus/spring/data/rest/deployment/SpringDataRestProcessor.java

```diff
@@ -0,0 +1,138 @@
+import io.quarkus.rest.data.panache.deployment.RestDataResourceBuildItem;
+import io.quarkus.rest.data.panache.deployment.properties.ResourcePropertiesBuildItem;
+import io.quarkus.spring.data.rest.deployment.crud.CrudMethodsImplementor;
+import io.quarkus.spring.data.rest.deployment.crud.CrudPropertiesProvider;
+import io.quarkus.spring.data.rest.deployment.paging.PagingAndSortingMethodsImplementor;
+import io.quarkus.spring.data.rest.deployment.paging.PagingAndSortingPropertiesProvider;
+
+class SpringDataRestProcessor {
+
+    private static final DotName CRUD_REPOSITORY_INTERFACE = DotName.createSimple(CrudRepository.class.getName());
+
+    private static final DotName PAGING_AND_SORTING_REPOSITORY_INTERFACE = DotName
+            .createSimple(PagingAndSortingRepository.class.getName());
+
+    private static final DotName JPA_REPOSITORY_INTERFACE = DotName.createSimple(JpaRepository.class.getName());
+
+    private static final List<DotName> EXCLUDED_INTERFACES = Arrays.asList(
+            CRUD_REPOSITORY_INTERFACE,
+            PAGING_AND_SORTING_REPOSITORY_INTERFACE,
+            JPA_REPOSITORY_INTERFACE);
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(SPRING_DATA_REST);
+    }
+
+    @BuildStep
+    void registerCrudRepositories(CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> implementationsProducer,
+            BuildProducer<RestDataResourceBuildItem> restDataResourceProducer,
+            BuildProducer<ResourcePropertiesBuildItem> resourcePropertiesProducer,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeansProducer) {
+        IndexView index = indexBuildItem.getIndex();
+
+        implementResources(implementationsProducer, restDataResourceProducer, resourcePropertiesProducer,
+                unremovableBeansProducer, new CrudMethodsImplementor(index), new CrudPropertiesProvider(index),
+                getRepositoriesToImplement(index, CRUD_REPOSITORY_INTERFACE));
+    }
+
+    @BuildStep
+    void registerPagingAndSortingRepositories(CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> implementationsProducer,
+            BuildProducer<RestDataResourceBuildItem> restDataResourceProducer,
+            BuildProducer<ResourcePropertiesBuildItem> resourcePropertiesProducer,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeansProducer) {
+        IndexView index = indexBuildItem.getIndex();
+
+        implementResources(implementationsProducer, restDataResourceProducer, resourcePropertiesProducer,
+                unremovableBeansProducer, new PagingAndSortingMethodsImplementor(index),
+                new PagingAndSortingPropertiesProvider(index),
+                getRepositoriesToImplement(index, PAGING_AND_SORTING_REPOSITORY_INTERFACE, JPA_REPOSITORY_INTERFACE));
+    }
+
+    private void implementResources(BuildProducer<GeneratedBeanBuildItem> implementationsProducer,
```

Comment:
I'd like to have a comment on this method describing what steps it performs and why they are needed